shell/platform/embedder/embedder_task_runner.cc

```diff
@@ -0,0 +1,72 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/embedder/embedder_task_runner.h"
+
+#include "flutter/fml/message_loop_impl.h"
+
+namespace shell {
+
+EmbedderTaskRunner::EmbedderTaskRunner(DispatchTable table)
+    : TaskRunner(nullptr /* loop implemenation*/),
+      dispatch_table_(std::move(table)) {
+  FML_DCHECK(dispatch_table_.post_task_callback);
+  FML_DCHECK(dispatch_table_.runs_task_on_current_thread_callback);
+}
+
+EmbedderTaskRunner::~EmbedderTaskRunner() = default;
+
+void EmbedderTaskRunner::PostTask(fml::closure task) {
+  PostTaskForTime(task, fml::TimePoint::Now());
+}
+
+void EmbedderTaskRunner::PostTaskForTime(fml::closure task,
+                                         fml::TimePoint target_time) {
+  if (!task) {
+    return;
+  }
+
+  uint64_t baton = 0;
+
+  {
+    // Release the lock before the jump via the dispatch table.
+    std::lock_guard<std::mutex> lock(tasks_mutex_);
+    baton = ++last_baton_;
+    pending_tasks_[baton] = task;
+  }
+
+  dispatch_table_.post_task_callback(this, baton, target_time);
```

Comment:
target_time values are based on the clock used by `fml::TimePoint::Now()`.  The embedder will need to expose an API that returns the value of that clock.

Reply:
I added FlutterEngineGetCurrentTime